test/CadlRanchProjects/type/model/inheritance/nested-discriminator/src/Generated/Models/Shark.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace _Type.Model.Inheritance.NestedDiscriminator.Models
+{
+    /// <summary>
+    /// The second level model in polymorphic multiple levels inheritance and it defines a new discriminator.
+    /// Please note <see cref="Shark"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+    /// The available derived classes include <see cref="GoblinShark"/> and <see cref="SawShark"/>.
+    /// </summary>
+    public partial class Shark : Fish
+    {
+        /// <summary> Initializes a new instance of <see cref="Shark"/>. </summary>
+        /// <param name="age"></param>
+        public Shark(int age) : base(age)
+        {
+            Sharktype = "shark";
```

Comment:
The reason why we said .Net generator is not fully support "nested discriminator" is here: this class is not correct.

According to the spec [here](https://github.com/Azure/cadl-ranch/blob/eb9e34f6a00bf2df3f8a9ce2efdc286698cb8d8c/packages/cadl-ranch-specs/http/type/model/inheritance/nested-discriminator/main.tsp#L19), the `Shark` class should have a discriminator `sharktype` defined, and have a property `kind` with the value of `shark`, but the generated code is messing them up together. Also because `Shark` defines a new discriminator `sharktype` but does not provide a value for `sharktype`, and because discriminator properties are required in payload, this class should be abstract to ensure that it is never instantiated because it will never have a proper value for `sharktype`.

We need to fix this first before we introduce this test case.

Reply:
```suggestion
            Kind = "shark";
```